smart-home-provider/cloud/auth-provider.js

```diff
@@ -304,6 +302,53 @@ Auth.registerAuth = function(app) {
 // &grant_type=authorization_code
 
 
+/**
+ * Decode authorization into client_id and client_secret.
+ * @return {{}}
+ * {
+ *   client_id: "CLIENT_ID",
+ *   client_secret: "CLIENT_SECRET"
+ * }
+ */
+function getHeaderAuthorization(req){
+  let headerAuthor = {};
+    try{
+      if(!!req.headers.authorization){
+        let authorizationString = Buffer.from(req.headers.authorization.split(" ")[1], 'base64').toString();
+        if(!!authorizationString){
+          let authorizationArray = authorizationString.split(':');
+          headerAuthor = {
+            client_id: authorizationArray[0],
+            client_secret: authorizationArray[1]
+          }
+        }
+      }
+    }catch(e){
+      return {};
+    }
+    return headerAuthor;
+}
+
+/**
+ * Get client's key from the following order:
```

Comment:
Did you run the linter on this commit? I think your functions may need to include `@param` doc strings.

Reply:
Okay.